lib/ansible/plugins/action/include_vars.py

```diff
@@ -40,12 +40,14 @@ def _set_dir_defaults(self):
 
         if isinstance(self.ignore_files, string_types):
             self.ignore_files = self.ignore_files.split()
+            self._display.deprecated(
+                msg="Specifying 'ignore_files' as a string is deprecated.",
+                version="2.23",
+                help_text="Use a list of strings instead.",
```

Comment:
Provide an `obj` reference so context can be provided to the user in the error message.

```suggestion
                help_text="Use a list of strings instead.",
                obj=self.ignore_files,
```

NOTE: You'll also need to move the deprecation before the assignment so `obj` is correct for the deprecation.